Northstar.CustomServers/mod/scripts/vscripts/_loadouts_mp.gnut

```diff
@@ -78,6 +85,21 @@ void function DelayDestroyDroppedWeapon( entity weapon )
 		weapon.Destroy()
 }
 
+void function WeaponDestroyThink()
+{
```

Comment:
`if( !file.weaponDropsEnabled )` should be here instead of continuously running a while-loop even if weapons drops are enabled.

Reply:
what if we change `file.weaponDropsEnabled` mid-game? I'll leave a commentary on it

Reply:
```suggestion
{
    if( !file.weaponDropsEnabled )
        thread TryAlwaysDestroyDroppedWeapon_Threaded()
}
```

Reply:
> what if we change `file.weaponDropsEnabled` mid-game? I'll leave a commentary on it

what if you made `WeaponDestroyThink()` a global function so gamemodes could just call it if they need to
probably register some signals too for ending the loop i guess.

Reply:
> > what if we change `file.weaponDropsEnabled` mid-game? I'll leave a commentary on it
> 
> what if you made `WeaponDestroyThink()` a global function so gamemodes could just call it if they need to probably register some signals too for ending the loop i guess.

that's a good idea, but my opinion is bind destroying dropped weapons together with `SetWeaponDropsEnabled( false )`.
executing will drop weapon is because they have a  "AE_WPNDROP" in their anim event, which is called earlier than victim's death, `AddCallback_OnPlayerKilled` can't handle it so we need a loop to keep destroying weapons

Reply:
> that's a good idea, but my opinion is bind destroying dropped weapons together with `SetWeaponDropsEnabled( false )`.
> executing will drop weapon is because they have a "AE_WPNDROP" in their anim event, which is called earlier than victim's death, `AddCallback_OnPlayerKilled` can't handle it so we need a loop to keep destroying weapons

Yeah I think that works best. You could register a signal in the same file like `RegisterSignal( "WeaponDropDisable" )` and just end the loop by signaling that.

Reply:
> > that's a good idea, but my opinion is bind destroying dropped weapons together with `SetWeaponDropsEnabled( false )`.
> > executing will drop weapon is because they have a "AE_WPNDROP" in their anim event, which is called earlier than victim's death, `AddCallback_OnPlayerKilled` can't handle it so we need a loop to keep destroying weapons
> 
> Yeah I think that works best. You could register a signal in the same file like `RegisterSignal( "WeaponDropDisable" )` and just end the loop by signaling that.

done that, is it a good idea to add new functions maybe `AlwaysDestroyDroppedWeaponEnabled()` to handle it?

Reply:
> > > that's a good idea, but my opinion is bind destroying dropped weapons together with `SetWeaponDropsEnabled( false )`.
> > > executing will drop weapon is because they have a "AE_WPNDROP" in their anim event, which is called earlier than victim's death, `AddCallback_OnPlayerKilled` can't handle it so we need a loop to keep destroying weapons
> > 
> > 
> > Yeah I think that works best. You could register a signal in the same file like `RegisterSignal( "WeaponDropDisable" )` and just end the loop by signaling that.
> 
> done that, is it good to add new functions maybe `AlwaysDestroyDroppedWeaponEnabled()` to handle it?

Doesn't hurt to have, I suppose.